### PR TITLE
Create Ansible task to install devspace operator cr

### PR DESCRIPTION
**Pull request by rocrisp**
#28

### CODE REVIEW

**Comment by rocrisp** [comment]
I need to add a task to clean up the operand.

**Review by bcrochet — CHANGES_REQUESTED** [review]
See comment

File: devspaces/tasks/main.yml [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+        - "'status' in checluster_info.resources[0]"
+        - "'chePhase' in checluster_info.resources[0].status"
+        - checluster_info.resources[0].status.chePhase == "Active" 
+  rescue:
+    - name: Print when failed
+      ansible.builtin.debug:
+        msg: "CheCluster Dashboard failed. Err: {{ ansible_failed_result }}"
+    - name: Register failed tests
+      ansible.builtin.set_fact:
+        test_result: "failed"
```
[the remark this answers is not in the document]
Comment (rocrisp) [reply devspaces/tasks/main.yml:49]: Hmm, upon checking, Matt uses "failed". See -> https://github.com/opdev/fips-assessments/blob/main/datagrid/tasks/main.yml#L141

[the remark this answers is not in the document]
Comment (rocrisp) [reply devspaces/tasks/main.yml:49]: Let me know what you guys decide

[the remark this answers is not in the document]
Comment (rocrisp) [reply devspaces/tasks/main.yml]: I just saw the final verdict from failed to fail

[the remark this answers is not in the document]
Comment (rocrisp) [reply devspaces/tasks/main.yml]: I made the change from failed to fail.

File: devspaces/tasks/main.yml [hunk 2]
```diff
@@ -0,0 +1,74 @@
+---
+- name: Setting Up Results Var
+  ansible.builtin.set_fact:
+    test_result: "pass"
```
Comment (bcrochet) [inline devspaces/tasks/main.yml:4]:
```suggestion
    assessment_result: "pass"
```

File: devspaces/tasks/main.yml [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+        name: example-checluster
+        api_version: org.eclipse.che/v2
+        kind: CheCluster
+        wait: true
+      register: checluster_info
+      retries: 30
+      delay: 50
+      until:
+        - checluster_info.resources | length > 0
+        - "'status' in checluster_info.resources[0]"
```
Comment (bcrochet) [inline devspaces/tasks/main.yml:40]:
```suggestion
        - checluster_info.resources[0].status is defined
```

File: devspaces/tasks/main.yml [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+        - "'chePhase' in checluster_info.resources[0].status"
```
Comment (bcrochet) [inline devspaces/tasks/main.yml:41]:
```suggestion
        - checluster_info.resources[0].status.chePhase is defined"
```

File: devspaces/tasks/main.yml [hunk 5]
```diff
@@ -0,0 +1,74 @@
...
+        - checluster_info.resources[0].status.chePhase == "Active"
```
Comment (bcrochet) [inline devspaces/tasks/main.yml:42]: I like how you grouped these!

File: devspaces/tasks/main.yml [hunk 6]
```diff
@@ -0,0 +1,74 @@
...
+        - "'status' in checluster_info.resources[0]"
+        - "'chePhase' in checluster_info.resources[0].status"
+        - checluster_info.resources[0].status.chePhase == "Active" 
+  rescue:
+    - name: Print when failed
+      ansible.builtin.debug:
+        msg: "CheCluster Dashboard failed to install. Err: {{ ansible_failed_result }}"
+    - name: Register failed tests
+      ansible.builtin.set_fact:
+        test_result: "fail"
```
Comment (bcrochet) [inline devspaces/tasks/main.yml:49]:
```suggestion
        assessment_result: "fail"
```

File: devspaces/tasks/main.yml [hunk 7]
```diff
@@ -0,0 +1,74 @@
...
+- name: Create result
```
Comment (bcrochet) [inline devspaces/tasks/main.yml:50]: If you change the above you can get rid of this task.

File: devspaces/tasks/main.yml [hunk 8]
```diff
@@ -0,0 +1,74 @@
...
+        kind: CheCluster
+        state: absent
+        wait: true
+  rescue:
+    - name: Print when failed
+      ansible.builtin.debug:
+        msg: "Cleanup failed. Err: {{ ansible_failed_result }}"
+    - name: Register failed tests
+      ansible.builtin.set_fact:
+        test_result: "fail"
```
Comment (bcrochet) [inline devspaces/tasks/main.yml:70]: I don't think you want to fail on cleanup. I'd remove this task.

File: devspaces/tasks/main.yml [hunk 9]
```diff
@@ -0,0 +1,74 @@
...
+# print pass or failed according to test_result fact variable
+- name: DevSpace Results
+  ansible.builtin.debug:
+    msg: "{% if test_result == 'pass' %} PASS {% else %} FAILED {% endif %}"
```
Comment (bcrochet) [inline devspaces/tasks/main.yml:74]:
```suggestion
    msg: "{% if assessment_result == 'pass' %} PASS {% else %} FAIL {% endif %}"
```
Alternatively (which is a little more idiomatic Ansible):

```suggestion
    msg: "{{ (assessment_result == 'pass') | ternary('PASS', 'FAIL') }}"
```

**Review by bcrochet — CHANGES_REQUESTED** [review]
A few comments and suggestions. Otherwise, looks good!

**Review by bcrochet — CHANGES_REQUESTED** [review]
Please squash to a single commit.

File: devspaces/tasks/main.yml (hunk 5 above)
Comment (rocrisp) [reply devspaces/tasks/main.yml:42]: Thanks

File: devspaces/tasks/main.yml [hunk 10]
```diff
@@ -0,0 +1,68 @@
...
+        state: absent
+        wait: true
+  rescue:
+    - name: Print when failed
+      ansible.builtin.debug:
+        msg: "Cleanup failed. Err: {{ ansible_failed_result }}"
+# print pass or failed according to test_result fact variable
+- name: DevSpace Results
+  ansible.builtin.debug:
+    msg: "{% if assessment_result == 'pass' %} PASS {% else %} FAIL {% endif %}"
```
Comment (bcrochet) [inline devspaces/tasks/main.yml:68]: File should have a trailing newline here.

**Review by bcrochet — CHANGES_REQUESTED** [review]
Just one more fix, collapse back to 1 commit, and this looks ready. Thanks!

**Review by bcrochet — APPROVED** [review]
/lgtm

**Pull request merged**